src/node.cc

```diff
@@ -2252,17 +2256,24 @@ void SetupProcessObject(Environment* env,
   env->SetMethod(process, "_getActiveHandles", GetActiveHandles);
   env->SetMethod(process, "_kill", Kill);
 
-  env->SetMethod(process, "cwd", Cwd);
+  env->SetMethod(process, "cwd", Cwd,
+                 SideEffectType::kHasNoSideEffect);
   env->SetMethod(process, "dlopen", DLOpen);
   env->SetMethod(process, "reallyExit", Exit);
-  env->SetMethod(process, "uptime", Uptime);
+  env->SetMethod(process, "uptime", Uptime,
+                 SideEffectType::kHasNoSideEffect);
 
 #if defined(__POSIX__) && !defined(__ANDROID__) && !defined(__CloudABI__)
-  env->SetMethod(process, "getuid", GetUid);
-  env->SetMethod(process, "geteuid", GetEUid);
-  env->SetMethod(process, "getgid", GetGid);
-  env->SetMethod(process, "getegid", GetEGid);
-  env->SetMethod(process, "getgroups", GetGroups);
+  env->SetMethod(process, "getuid", GetUid,
```

Comment:
Just a thought... but a `env->SetSafeMethod(process, ...)` variant that sets `SideEfectType::kHasNoSideEffect` automatically may be worthwhile.